mkosi/__init__.py

```diff
@@ -2704,7 +2739,9 @@ def install_arch(args: MkosiArgs, root: Path, do_run_build_script: bool) -> None
     if args.release is not None:
         MkosiPrinter.info("Distribution release specification is not supported for Arch Linux, ignoring.")
 
-    if args.mirror:
+    if args.plain_mirror and args.mirror:
+        server = f"Server = {args.mirror}"
```

Comment:
I haven't run this (and I'm not quite sure how to test this properly), but I'm a bit unsure about the semantics here. This omits the `$repo/os/$arch` trailer on the `Server` directive so that pacman would look for `core`, `extra` and `community` in the very same location, i.e. not interpolate the names of the repos (and first architecture), so that all three of them would need to be same repo. This necessitates some custom merged repo setup instead of just some mirrors, because we install packages from both `core` and `extra` (mainly dracut), no?

Reply:
I have not tested this on arch yet, only on rpm ones

Reply:
Might be safer to drop this for a first go then, if nobody else with more knowledge of pacman repo management innards interjects (@Foxboron maybe?).

Reply:
It should just work™️. But I find this weird and I'm not sure how OBS deals with Arch repositories? Is there any docs on the offline builds somewher?

It's partially a bikeshed, but this is effectively just an online package cache with the database files, not a "plain mirror". A "flat mirror" would a better name I think. But I'll leave that up for the maintainers :)

Reply:
I have not yet implemented it, but it will do the same thing that kiwi does, calling repo-add on the directory where the packages are downloaded in a separate step by the resolver:

https://github.com/bluca/obs-build/blob/mkosi/build-recipe-kiwi#L812

I'll remove the arch part for now

Reply:
Hmm, okay. But this won't give you `core`, `extra`, `community`. It would give you one repo defined to be whatever the name of the database is. If you want to make 3 repos in the same directory you'd need to specify each of the packages manually instead of doing a glob.

This implies that the `pacman.conf` as distributed by `mkosi` is not going to be compatible with the flat repository you are currently building there.

So yeah, I think removing the pacman part of this is a good idea :)

Reply:
Yes, that is how it works, all the actual repositories (in all their multiple forms) are configured on the source project and fetched by the resolver in the normal way, when the dependency resolution step for a given build is performed. This on the other hand is the isolated build step - the scheduler gets all the packages that were downloaded for a given build job, and makes them available in a local directory to the build worker root, which is booted in kvm with a custom init that just installs them all and calls the build hooks.
So at the build stage you don't have multiple repositories or multiple definitions or anything like that, because it's moot - it's completely offline with no network access whatsoever by design.